tests/phpunit/tests/query/cacheResults.php

```diff
@@ -1211,6 +1211,41 @@ public function data_query_cache_with_empty_result_set() {
 		);
 	}
 
+	/**
+	 * Ensure post meta/term caches are warmed even if a post is already cached.
+	 *
+	 * @covers ::_prime_post_caches
+	 *
+	 * @ticket 57163
+	 */
+	public function test_post_meta_and_term_caches_are_warmed_even_if_post_is_already_cached() {
```

Comment:
Not sure if this is a good test but at least a start since it fails without the changes.